### ISSUE TITLE
PayPal Subscriptions

### ISSUE THREAD

**Issue by lennartlott** [craftcms/commerce-paypal-checkout#29]
Any plans to work with PayPal subscriptions anytime soon?

**Comment by nfourtythree** [comment]
Hi @lennartlott 

This is something we are aware of and something we would like to look into in the future. Currently, we do not have a specific timeline for it.

We are always open for people to submit PRs to the plugin for adding new features.

Thanks!

**Comment by lennartlott** [comment]
@nfourtythree Can you give me a price for this feature?

**Comment by richardcalahan** [comment]
Any updates on this feature?

**Comment by mufi-siebird** [comment]
Its been months. Is this still in pipeline?

**Comment by nfourtythree** [comment]
Hi @mufi-siebird 

As my previous comment, this is still a feature request but we do not have a specific timeline as to when this feature will be implemented.

Thanks.

**Comment by klick** [comment]
Hey everyone. 
Any news on this?

**Comment by mufi-siebird** [comment]
> Hey everyone. Any news on this?

2 years ago, we have created custom payment gateway to have subscriptions with Paypal. I dont recommend that way but I guess thats the only valid option as of today.

**Comment by klick** [comment]
> > Hey everyone. Any news on this?
> 
> 2 years ago, we have created custom payment gateway to have subscriptions with Paypal. I dont recommend that way but I guess thats the only valid option as of today.

Seems so. Is there any documentation on how to create a custom gateway? Can’t seem to find anything.

**Comment by mufi-siebird** [comment]
> > > Hey everyone. Any news on this?
> > 
> > 
> > 2 years ago, we have created custom payment gateway to have subscriptions with Paypal. I dont recommend that way but I guess thats the only valid option as of today.
> 
> Seems so. Is there any documentation on how to create a custom gateway? Can’t seem to find anything.

No. I created it by digging around craft commerce and paypal checkout plugin code.

**Comment by klick** [comment]
Too bad. Care to share your code for me to dig around as a starting point? Seems that not too many are interested in PayPal subscriptions who also run a Craft Commerce store.

**Comment by klick** [comment]
> No. I created it by digging around craft commerce and paypal checkout plugin code.

Too bad. Care to share your code for me to dig around as a starting point? Seems that not too many are interested in PayPal subscriptions who also run a Craft Commerce store.

**Comment by mufi-siebird** [comment]
> > No. I created it by digging around craft commerce and paypal checkout plugin code.
> 
> Too bad. Care to share your code for me to dig around as a starting point? Seems that not too many are interested in PayPal subscriptions who also run a Craft Commerce store.

Sure. whats your email?

**Comment by klick** [comment]
> > > No. I created it by digging around craft commerce and paypal checkout plugin code.
> > 
> > 
> > Too bad. Care to share your code for me to dig around as a starting point? Seems that not too many are interested in PayPal subscriptions who also run a Craft Commerce store.
> 
> Sure. whats your email?

Cool. Here ist is: desk@marcusscheller.com
That would be great!

**Comment by pk-nea** [comment]
> Seems that not too many are interested 

seems like here is one of those few. 
would it be possible for either of you klick or mufi to let me have a look at the code as well?
i'd really appreciate it.

**Comment by mufi-siebird** [comment]
Sure. I am making it open for now.
https://www.dropbox.com/sh/jt83tuuommaz944/AADWMgAaHZUFJs_fDgMmYjy_a?dl=0

**Comment by pk-nea** [comment]
thank you!
